interfaces/cython/cantera/thermo.pyx

```diff
@@ -678,6 +678,19 @@ cdef class ThermoPhase(_SolutionBase):
             else:
                 self._setArray1(thermo_setMassFractions, Y)
 
+    property Y_no_norm:
```

Comment:
As I said in another comment, I think you can be more explicit with the code and get around needing to add these two properties. I don't think that these will be used very much in general, and I don't particularly want to make it easier for people to set un-normalized mass or mole fractions. It is almost never what the user actually wants, so making it easier to access doesn't seem great to me. Finally, this is totally personal preference, but all of our single-letter setters are only available as the single letter (I'm pretty sure), so this would break that paradigm.

Reply:
This setter was based on my suggestion, where the naming is loosely based on C++'s `setMassFractions_NoNorm`. The rationale here was to implement an efficient way to reach the C++ layer for the non-normalized setters that this PR seeks to implement.

Regarding single-letter (or multi-letter) uppercase setters (or getters), I wasn't aware of a paradigm (other than them not being pythonic, where the rationale is obvious). There isn't a `setMassFractions` setter in Python, so the suggested property is imho closest to what is already there. Regarding the concern of making this available to users, it would be simple to hide (via `_Y_no_norm` etc.), but at the same time there may be an argument to allow this for those users who need it and locate it in the documentation.

Reply:
> This setter was based on my suggestion, where the naming is loosely based on C++'s `setMassFractions_NoNorm`. The rationale here was to implement an efficient way to reach the C++ layer for the non-normalized setters that this PR seeks to implement.

The non-normalized setter is already implemented in Python, https://github.com/Cantera/cantera/blob/627e21e95e2dc534dabf9fb8fda1e4093177704e/interfaces/cython/cantera/thermo.pyx#L944-L970, which this implementation just calls. So I don't think these new properties are needed, and the existing functions can be used directly.

Reply:
Good point - this was quite obviously hiding in plain sight.